src/Bicep.Core/Semantics/Namespaces/AzNamespaceSymbol.cs

```diff
@@ -54,6 +54,65 @@ private static ObjectType GetSubscriptionReturnValue(IEnumerable<FunctionArgumen
+                new TypeProperty("media", LanguageConstants.String),
+                new TypeProperty("sqlManagement", LanguageConstants.String),
+                new TypeProperty("vmImageAliasDoc", LanguageConstants.String),
+                new TypeProperty("resourceManager", LanguageConstants.String),
+                new TypeProperty("authentication", new NamedObjectType("authentication", TypeSymbolValidationFlags.Default, new []
+                {
+                    new TypeProperty("loginEndpoint", LanguageConstants.String),
+                    new TypeProperty("audiences", new TypedArrayType(LanguageConstants.String, TypeSymbolValidationFlags.Default)),
+                    new TypeProperty("tenant", LanguageConstants.String),
+                    new TypeProperty("identityProvider", LanguageConstants.String),
+                }, null)),
+                new TypeProperty("suffixes", new NamedObjectType("suffixes", TypeSymbolValidationFlags.Default, new []
+                {
+                    new TypeProperty("acrLoginServer", LanguageConstants.String),
+                    new TypeProperty("azureDatalakeAnalyticsCatalogAndJob", LanguageConstants.String),
+                    new TypeProperty("azureDatalakeStoreFileSystem", LanguageConstants.String),
+                    new TypeProperty("keyvaultDns", LanguageConstants.String),
+                    new TypeProperty("sqlServerHostname", LanguageConstants.String),
+                    new TypeProperty("storage", LanguageConstants.String),
+                }, null)),
+                new TypeProperty("locations", new TypedArrayType(new NamedObjectType("locations", TypeSymbolValidationFlags.Default, new []
+                {
+                    new  TypeProperty("id", LanguageConstants.String),
+                    new  TypeProperty("name", LanguageConstants.String),
+                    new  TypeProperty("displayName", LanguageConstants.String),
+                    new  TypeProperty("longitude", LanguageConstants.String),
+                }, null), TypeSymbolValidationFlags.Default)),
+            }, null);
+        }
+
+        private static NamedObjectType GetDeploymentReturnType(ResourceScopeType targetScope)
+        {
+            if (targetScope.HasFlag(ResourceScopeType.ResourceGroupScope))
+            {
+                // deployments in the 'resourcegroup' scope do not have the 'location' property. All other scopes do.
+                return new NamedObjectType("environment", TypeSymbolValidationFlags.Default, new []
+                {
+                    new TypeProperty("name", LanguageConstants.String),
+                }, null);
+            }
+
+            return new NamedObjectType("environment", TypeSymbolValidationFlags.Default, new []
+            {
+                new TypeProperty("name", LanguageConstants.String),
+                new TypeProperty("location", LanguageConstants.String),
+            }, null);
```

Comment:
Technically there are many other properties which may or may not appear, but skimming the quickstarts repo I couldn't find any common cases apart from templateLinks.Uri - which is not going to be meaningful in Bicep.